tsl/src/bgw_policy/job.c

```diff
@@ -770,7 +786,11 @@ job_execute(BgwJob *job)
 	if (!ActiveSnapshotSet())
 	{
 		pushed_snapshot = true;
+#if (PG12 && PG_VERSION_NUM >= 120008) || (PG13 && PG_VERSION_NUM >= 130004) || PG14_GE
```

Comment:
This is possible only if we have to setup a Portal, right? I think we can combine if conditions at 771, 779 and 786. If not, could you explain where else we might have to start a TransactionBlock ?

Reply:
Definitely we can do it. Fixed!